File: app/src/main/java/org/droidupnp/model/upnp/CallableContentDirectoryFilter.java
package org.droidupnp.model.upnp;

public class CallableContentDirectoryFilter implements ICallableFilter {

	private IUPnPDevice device;

	public void setDevice(IUPnPDevice device)
	{
		this.device = device;
	}

	@Override
	public Boolean call() throws Exception
	{
		return device.asService("ContentDirectory");
	}
}

File: app/src/main/java/org/droidupnp/legacy/upnp/CallableContentDirectoryFilter.kt
package org.droidupnp.legacy.upnp

import com.m3sv.plainupnp.data.UpnpDevice

class CallableContentDirectoryFilter : ICallableFilter {

    private var device: UpnpDevice? = null

    override fun setDevice(device: UpnpDevice) {
        this.device = device
    }

    override fun call(): Boolean? = device?.asService("ContentDirectory") ?: false
}
